### PR TITLE
raftstore: real batch ingest sst files in different region

### PR DESCRIPTION
**Pull request by 3pointer**
### What is changed and how it works?

Issue Number: ref https://github.com/tikv/tikv/issues/16267

What's Changed:

```commit-message
raftstore: real batch ingest sst files in different region
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [x] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
raftstore: batch apply ingest sst files with different regions.
```

### CODE REVIEW

File: components/raftstore/src/store/fsm/apply.rs [hunk 1]
```diff
@@ -2005,7 +2015,7 @@ where
                 panic!("{} ingest {:?}: {:?}", self.tag, sst, e);
             }
         };
-
+        self.has_pending_ssts = true;
```
Comment (glorv) [inline components/raftstore/src/store/fsm/apply.rs:2018]: NIT: better move to the Ok(..) branch

**Review by glorv — APPROVED** [review]
LGTM

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**